### PR TITLE
Add migration for `distinct_id` column on clickhouse person table

### PR DESCRIPTION
**Pull request by macobo**
Note that code under ee/clickhouse/models/person.py does not work
anymore since it does not populate the column correctly. That is fine -
will be handled when doing the actual migration, rather than for this
experiment.

Issue: https://github.com/PostHog/posthog/issues/5167

Depends on https://github.com/PostHog/plugin-server/pull/507 - should be deployed together with it.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

File: ee/clickhouse/migrations/0013_persons_distinct_ids_column.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from infi.clickhouse_orm import migrations
+
+from ee.clickhouse.sql.person import KAFKA_PERSONS_TABLE_SQL, PERSONS_TABLE_MV_SQL
+from posthog.settings import CLICKHOUSE_CLUSTER
+
+operations = [
+    migrations.RunSQL(f"DROP TABLE person_mv ON CLUSTER {CLICKHOUSE_CLUSTER}"),
+    migrations.RunSQL(f"DROP TABLE kafka_person ON CLUSTER {CLICKHOUSE_CLUSTER}"),
+    migrations.RunSQL(
+        f"ALTER TABLE person ON CLUSTER {CLICKHOUSE_CLUSTER} ADD COLUMN IF NOT EXISTS distinct_ids Array(VARCHAR)"
```
Comment (neilkakkar) [inline ee/clickhouse/migrations/0013_persons_distinct_ids_column.py:10]: Q: Does this work without a default? What happens to `distinct_ids` on pre-existing persons?

**Comment by neilkakkar** [comment]
I'm assuming that we'll also update *some* queries to use these distinct_ids on persons. Will that be a follow up PR? How will that work, assuming some Persons will have distinct_ids, and others won't ?

File: ee/clickhouse/migrations/0013_persons_distinct_ids_column.py (hunk 1 above)
Comment (macobo) [reply ee/clickhouse/migrations/0013_persons_distinct_ids_column.py:10]: They are not populated. Forgot to mention in the description that if the experiment is successful, we'll need to back-fill these values.

**Comment by macobo** [comment]
> I'm assuming that we'll also update some queries to use these distinct_ids on persons.

Nope. What I'll do is:
1. Wait for data to trickle in
2. Select a sampling of slow queries joining against person_distinct_id
3. Measure results in time range where the column has been populated
4. (Optional) Measure them with clickhouse-flamegraph
5. Run and measure the queries without the person_distinct_id join and instead using the new columns
6. Document under issue, decide how to proceed (kill vs solve issues + rewrite queries)

**Review by neilkakkar — APPROVED** [review]

**Pull request merged**